domain-models-interface-extensions/src/test/java/org/folio/rest/tools/ClientGeneratorTest.java

```diff
@@ -45,204 +45,21 @@ public void doesGenerateTestResourceClient() throws Exception {
-    expected.add("            queryParams.append(\"&\");");
-    expected.add("        }");
-    expected.add("        if (publicationDate!= null) {");
-    expected.add("            queryParams.append(\"publicationDate=\");");
-    expected.add("            queryParams.append(DateTimeFormatter.ISO_LOCAL_DATE_TIME.format(ZonedDateTime.ofInstant(publicationDate.toInstant(), ZoneId.of(\"UTC\"))));");
-    expected.add("            queryParams.append(\"&\");");
-    expected.add("        }");
-    expected.add("        if (rating!= null) {");
-    expected.add("            queryParams.append(\"rating=\");");
-    expected.add("            queryParams.append(rating);");
-    expected.add("            queryParams.append(\"&\");");
-    expected.add("        }");
-    expected.add("        if (isbn!= null) {");
-    expected.add("            queryParams.append(\"isbn=\");");
-    expected.add("            queryParams.append(URLEncoder.encode(isbn, \"UTF-8\"));");
-    expected.add("            queryParams.append(\"&\");");
-    expected.add("        }");
-    expected.add("        if (facets!= null) {");
-    expected.add("            queryParams.append(\"facets=\");");
-    expected.add("            if(facets.getClass().isArray()){queryParams.append(String.join(\"&facets=\",facets));}");
-    expected.add("            queryParams.append(\"&\");");
-    expected.add("        }");
-    expected.add("        io.vertx.core.http.HttpClientRequest request = httpClient.getAbs(okapiUrl+\"unittestsbooks\"+queryParams.toString());");
-    expected.add("        request.handler(responseHandler);");
-    expected.add("        request.putHeader(\"Accept\", \"application/json\");");
-    expected.add("        if (tenantId!= null) {");
-    expected.add("            request.putHeader(\"X-Okapi-Token\", token);");
-    expected.add("            request.putHeader(\"x-okapi-tenant\", tenantId);");
-    expected.add("        }");
-    expected.add("        if (okapiUrl!= null) {");
-    expected.add("            request.putHeader(\"X-Okapi-Url\", okapiUrl);");
-    expected.add("        }");
-    expected.add("        request.end();");
-    expected.add("    }");
-    expected.add("");
-    expected.add("}");
-
-    List<String> actual = FileUtils.readLines(expectedClient, StandardCharsets.UTF_8);
-
-    Assert.assertEquals(expected.size(), actual.size());
+    // IDEs always removes trailing spaces from edited files, but java code generator adds then,
+    // so we nee to remove them before the comparison
+
+    String actual = removeTrailingSpaces(FileUtils.readFileToString(expectedClient, StandardCharsets.UTF_8));
+
+    String expected = removeTrailingSpaces(FileUtils.readFileToString(
+        new File(ClientGeneratorTest.class.getResource("/clients/TestClient.txt").getFile()),
+        StandardCharsets.UTF_8));
+
     Assert.assertEquals(expected, actual);
   }
 
+
+  private String removeTrailingSpaces(String str) {
+    return str.replaceAll("(\\s+)\n","\n");
```

Comment:
Braces are not needed.
`\n` doesn't match all possible end of line characters, and it doesn't match the last line that does not end with `\n`.

    private static final Pattern TRAINLING_SPACE_PATTERN = Pattern.compile("\\s+$", Pattern.MULTILINE);
    
    private String removeTrailingSpaces(String str) {
      return TRAINLING_SPACE_PATTERN.matcher(str).replaceAll("");
    }

Reply:
done.